python/tank/bootstrap/manager.py

```diff
@@ -937,6 +967,57 @@ def _start_engine(self, tk, engine_name, entity, progress_callback=None):
 
         return engine
 
+    def _start_shotgun_engine(self, tk, entity, progress_callback=None):
+        """
+        Launch into the tk-shotgun engine using a legacy shotgun_*.yml environment.
+
+        If entity is None, the method will bootstrap into the site config.
+
+        Please note that the API version of the tk instance that hosts
+        the engine may not be the same as the API version that was
+        executed during the bootstrap.
+
+        :param tk: Bootstrapped :class:`~sgtk.Sgtk` instance.
+        :param entity: Shotgun entity used to resolve a project context.
+        :type entity: Dictionary with keys ``type`` and ``id``, or ``None`` for the site.
+        :param progress_callback: Callback function that reports back on the engine startup progress.
+                                  Set to ``None`` to use the default callback function.
+        :returns: Started :class:`~sgtk.platform.Engine` instance.
+        """
+
+        # Log a user activity metric saying which engine and entity are bootstrapping
+        # since we can now log into the logging queue of the new swapped core.
+        engine_name = "tk-shotgun"
```

Comment:
seeing this defined twice in both methods feels like another hint that we should merge the methods into one